src/pytest_cov/plugin.py

```diff
@@ -300,6 +302,11 @@ def pytest_runtest_call(self, item):
         else:
             yield
 
+    def pytest_runtest_logreport(self, report):
+        if self.options.cov_context:
+            context = "{0.nodeid}|{0.outcome}|{0.when}".format(report)
+            self.cov_controller.cov.switch_context(context)
```

Comment:
So it stores the context after the fact, i.e. for previous data?
I've read it differently (although it could be clearer in the changelog for example), and the code also says so: https://github.com/nedbat/coveragepy/blob/f9cb7becf69b299f5f946944d855b7e964afec8d/coverage/sqldata.py#L206-L211

So it looks like this should use (a) different hook(s): https://github.com/pytest-dev/pytest/blob/865e84d206581f77640cfbc1516aff7a0d7316c6/src/_pytest/hookspec.py#L347-L362

Reply:
Hmm, you're probably right that it's after the fact, that is, wrong :)